### PR TITLE
better logging for ARAKOON-422 (+ test)

### PR DESCRIPTION
**Pull request by toolslive**

### CODE REVIEW

**Comment by domsj** [comment]
LGTM

File: extension/test/server/right/system_tests_long_right.py [hunk 1]
```diff
@@ -560,3 +560,71 @@ def test_db_optimize():
...
+    nn = Common.node_names[0]
+    Common.stopOne(nn)
+    fn = Common.get_node_db_file(nn)
+    os.remove(fn)
+    logging.info("removed %s", fn)
+    cluster = _getCluster()
+    fs = Common.q.system.fs
+    cfg = cluster.getNodeConfig(nn)
+    node_tlf_dir  = cfg['tlf_dir']
+    tlf_full_path = fs.joinPaths (node_tlf_dir, "%03d.tlf" % 2)
```
Comment (NicolasT) [inline extension/test/server/right/system_tests_long_right.py]: `"002.tlf"` would be less odd.

File: extension/test/server/right/system_tests_long_right.py [hunk 2]
```diff
@@ -560,3 +560,71 @@ def test_db_optimize():
...
+            n = 0
+
+        if n == 0:
+            wait = False
+
+        t1 = time.time()
+        d = t1 - t0
+        logging.info("after d=%fs, the node's still running", d)
+        if d > 120.0:
+            wait = False
```
Comment (NicolasT) [inline extension/test/server/right/system_tests_long_right.py:604]: Throw some error instead?

File: extension/test/server/right/system_tests_long_right.py [hunk 3]
```diff
@@ -560,3 +560,71 @@ def test_db_optimize():
...
+    
+    if n:
+        assert_equals(True,False,"node still running")
```
Comment (NicolasT) [inline extension/test/server/right/system_tests_long_right.py]: I'd raise an error, there's no real assertion being done.

Or, alternatively (without an `if` block):

`assert_equals(n, 0, "Node should not be running")`

File: extension/test/server/right/system_tests_long_right.py [hunk 4]
```diff
@@ -560,3 +560,71 @@ def test_db_optimize():
...
+    f.seek(size - 4096)
+    tail = f.read()
+    f.close()
+    ok = False
+    for line in tail.split('\n'):
+        if line.find("(found neither 002.tlf nor 002.tlog)"):
+            logging.info("line=%s",line)
+            ok = True
+                     
+    assert_equals(True,ok,"line should be present")
```
Comment (NicolasT) [inline extension/test/server/right/system_tests_long_right.py]: Is there no `assert_(ok, "Line should be present")`?

File: src/node/catchup.ml [hunk 5]
```diff
@@ -239,11 +242,30 @@ let catchup_store (type s) me ((module S : Store.STORE with type t = s), store,t
         let pred_too_far_i = Sn.pred too_far_i in
         if si < pred_too_far_i
         then
-          let msg = Printf.sprintf
-                      "Catchup store failed. Store counter is too low: %s < %s"
-                      (Sn.string_of si) (Sn.string_of pred_too_far_i)
-          in
-          Lwt.fail (StoreCounterTooLow msg)
+	  begin
```
Comment (NicolasT) [inline src/node/catchup.ml]: I'm not sure ocp-indent agrees with this ;-)

File: src/tlog/tlc2.ml [hunk 6]
```diff
@@ -694,12 +694,29 @@ class tlc2 (tlog_dir:string) (tlf_dir:string) (head_dir:string) (new_c:int)
...
+    method which_tlog_file (start_i : Sn.t) = 
+      let n = Sn.to_int (get_file_number start_i) in
+      let an = archive_name n 
+      and fn = file_name n 
+      in
+      let an_c = get_full_path tlog_dir tlf_dir an in
+      let fn_c = get_full_path tlog_dir tlf_dir fn 
+      in
+      File_system.exists an_c >>= function
+	| true -> Lwt.return (Some an_c)
```
Comment (NicolasT) [inline src/tlog/tlc2.ml]: Same ocp-indent remark

**Pull request merged**